src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.core
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.config.ErrorResponse
+
+@RestController
+@Validated
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class DocumentResource(private val documentService: DocumentService) {
+  @PreAuthorize("hasRole('ROLE_NOMIS_DOCUMENTS')")
+  @GetMapping("/documents/{id}", produces = ["application/x-zip-compressed"])
```

Comment:
a weird content type but I guess there is a reason for that?

Reply:
wrong type

Reply:
will fix these in next PR